mlir/lib/Bindings/Python/IRModule.h

```diff
@@ -1124,6 +1122,8 @@ class PyValue {
   /// Gets a capsule wrapping the void* within the MlirValue.
   pybind11::object getCapsule();
 
+  virtual pybind11::object maybeDownCast();
```

Comment:
This was a fun "whodoneit"; if you want RTTI to work correctly for `ConcreteValue` i.e., you want pybind to be able to figure out the correct thing to `py::cast` to (`OpResult` or `BlockArgument`), then you need at least [one virtual member](https://stackoverflow.com/a/11484105). Thanks @zero9178 for the tip!

Reply:
The tradition is to make a virtual destructor...  Do need `py::cast` to work for these? I'd rather not pay for something we don't use. It looks like this would be necessary if caster functions wanted to take BlockArgument or OpResult specifically, but arguably they should always take Value and can do `isa` inside if really needed.

Reply:
> Do need py::cast to work for these?

You need the `py::cast` in there so that you're returning a single type `py::object` (like the casters return) instead of `py::object | PyValue`. But it's moot because it works with the virtual destructor just the same.